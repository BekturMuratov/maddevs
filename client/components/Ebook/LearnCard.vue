<template>
  <section class="ebook-learn-card">
    <img
      v-if="icon"
      :src="require(`@/assets/img/Ebook/${icon}`)"
      alt="Icon"
    >
    <h4>{{ title }}</h4>
    <p>{{ description }}</p>
  </section>
</template>

<script>
export default {
  name: 'EbookLearnCard',
  props: {
    icon: {
      type: String,
      default: null,
    },

    title: {
      type: String,
      default: 'Card title',
    },

    description: {
      type: String,
      default: 'Card description',
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

.ebook-learn-card {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  background-color: #1D1D1F;
  border-radius: 4px;
  padding: 24px;

  img {
    width: 54px;
    height: 54px;
    margin-bottom: 27px;
  }

  h4 {
    @include font('Inter', 21px, 600);
    line-height: 24px;
    letter-spacing: -1px;
    color: #fff;
    margin-bottom: 21px;
  }

  p {
    @include font('Inter', 16px, 400);
    line-height: 24px;
    letter-spacing: -0.4px;;
    color: #CFCFD0;
  }
}
</style>
