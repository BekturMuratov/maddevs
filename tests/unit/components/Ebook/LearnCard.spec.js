import 'regenerator-runtime'
import { shallowMount } from '@vue/test-utils'
import LearnCard from '@/components/Ebook/LearnCard'

describe('LearnCard component', () => {
  let wrapper = null

  beforeEach(() => {
    wrapper = shallowMount(LearnCard)
  })

  it('should render correctly with no data', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
