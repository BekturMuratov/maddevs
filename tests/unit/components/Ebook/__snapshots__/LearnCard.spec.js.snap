// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnCard component should render correctly with no data 1`] = `
<section class="ebook-learn-card">
  <!---->
  <h3 class="ebook-learn-card_title">
    Card title
  </h3>
  <p class="ebook-learn-card_desc">
    Card description
  </p>
</section>
`;
